bitnami/seaweedfs/templates/_helpers.tpl

```diff
@@ -77,6 +77,19 @@ Return the proper init external database job image name
 {{- end -}}
 {{- end -}}
 
+{{/*
+Returns whether wait for external database is enabled
+*/}}
+{{- define "seaweedfs.waitForDatabase.enabled" -}}
+{{- if or (and .Values.mariadb.enabled (not .Values.externalDatabase.enabled) ) (and .Values.externalDatabase.enabled (eq .Values.externalDatabase.store "mariadb") (eq .Values.externalDatabase.waitForDatabaseEnabled true) ) }}
+    {{- print "true" -}}
+  {{- else if or (and .Values.postgresql.enabled (not .Values.externalDatabase.enabled) ) (and .Values.externalDatabase.enabled (eq .Values.externalDatabase.store "postgresql") (eq .Values.externalDatabase.waitForDatabaseEnabled true) ) }}
+    {{- print "true" -}}
+  {{- else }}
+    {{- print "false" -}}
+{{- end -}}
+{{- end -}}
+
```

Comment:
It seems not necessary to check the external DB if mariadb or postgresql are enabled.
```suggestion
{{- if or .Values.mariadb.enabled (and .Values.externalDatabase.enabled (eq .Values.externalDatabase.store "mariadb") (eq .Values.externalDatabase.waitForDatabaseEnabled true) ) }}
    {{- print "true" -}}
  {{- else if or .Values.postgresql.enabled (and .Values.externalDatabase.enabled (eq .Values.externalDatabase.store "postgresql") (eq .Values.externalDatabase.waitForDatabaseEnabled true) ) }}
    {{- print "true" -}}
  {{- else }}
    {{- print "false" -}}
{{- end -}}
{{- end -}}

```

Let me know if I'm wrong

Reply:
@dgomezleon : I just pushed simplified logic based on your feedback.  Please review the latest.